app/assets/javascripts/admin/views/project/project_list_view.js

```diff
@@ -26,23 +27,39 @@ type StateProps = {
   activeUser: APIUserType,
 };
 
+type Props = {
+  history: RouterHistory,
+} & StateProps;
+
 type State = {
   isLoading: boolean,
   projects: Array<APIProjectType>,
   searchQuery: string,
 };
 
-class ProjectListView extends React.PureComponent<StateProps, State> {
+const STATE_TO_BE_PERSISTED: { [$Keys<State>]: Function } = {
```

Comment:
Since you are planning to use this in several places, we might want to try to make this even more concise. How about the following as an interface?

```
const persistence: Persistence<State> = new Persistence({
    searchQuery: PropTypes.string,
}, "projectList");

persistence.load(this.props.history);

persistence.persist(this.props.history, nextState);
```

The class could look something like [this](0C4TwDgpgBAysCGxoF4oG8BQUoEsDOAMgPbwAmOAdgOYBcUARkUQDYTwUA0WUSewdfAE6UqXAL4BuDBgDGzeHjxQAChEF58SCjIgAeGAD503AJAV4AWwgDgw6lJN9EEACpEAQhFXrNEUnQB5egArCBlgKWwMExkiCiEAV3CiQQAKaMcEJDdPbw0).

Reply:
@philippotto Yep, I like that :) I cannot see your suggestion for the class though, Github shows a Not Found page :/

Reply:
Sorry about that; I fixed the link in the upper comment now :)